include/dd/Package.hpp

```diff
@@ -579,11 +575,11 @@ namespace dd {
             auto       z             = static_cast<Qubit>(start);
             const auto smallerTarget = std::min(target0, target1);
             for (; z < smallerTarget; ++z) {
```

Comment:
This whole function can be simplified quite a lot. As there are no controls, it simply constructs a two-level DD with the appropriate nodes. All the special handling below, in-between, and above targets can be eliminated.

Actually, at that point (also relating to the comment above), this basically becomes the routine that takes a matrix and constructs the DD. Maybe that can also be reused here or used instead of the method here.

Reply:
I simplified the function like you mentioned, so now it only directly affects the targets.

I tried using the function makeDDFromMatrix to simplify it. First, I overloaded it to take the matrix of ComplexValue as input rather than just a CMat.

I'll leave this as a TODO for now, since it is currently working at least and slightly out of the scope of this PR.